classes/security/authorization/SubmissionFileAccessPolicy.inc.php

```diff
@@ -184,9 +211,11 @@ function buildFileAccessPolicy($request, $args, $roleAssignments, $mode, $fileId
 			$subEditorFileAccessPolicy = new PolicySet(COMBINING_DENY_OVERRIDES);
 			$subEditorFileAccessPolicy->addPolicy(new RoleBasedHandlerOperationPolicy($request, ROLE_ID_SUB_EDITOR, $roleAssignments[ROLE_ID_SUB_EDITOR]));
 
-			// 2) ... but only if they have been assigned to the requested submission.
+			// 2) ... but only if they have been assigned as a subeditor to the requested submission.
 			import('lib.pkp.classes.security.authorization.internal.UserAccessibleWorkflowStageRequiredPolicy');
 			$subEditorFileAccessPolicy->addPolicy(new UserAccessibleWorkflowStageRequiredPolicy($request));
```

Comment:
I don't think it's necessary here. Having looked at it, it seems redundant. We only call the `UserAccessibleWorkflowStageRequiredPolicy` here so that we have the `ASSOC_TYPE_ACCESSIBLE_WORKFLOW_STAGES` in the authorized context objects. The `AssignedStageRoleHandlerOperationPolicy` should provide the only check we need.

I _think_ we could add `WorkflowStageAccessPolicy` as well without causing any harm, but I am a bit nervous. The `WorkflowStageAccessPolicy` relies on the overall roles in a context, not the assigned role in the submission, so I'm not 100% that it is providing a useful check here.

It would also lead to extra database reads. (`SubmissionRequiredPolicy` will be called twice.)

:man_shrugging: